File: leetcode/1351. Count Negative Numbers in a Sorted Matrix/1351.hs
-- https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
countNegatives :: [[Int]] -> Int
countNegatives = length . filter (<0) . concat 

-- Tests
main :: IO()
main = do
    print $ countNegatives [[4,3,2,-1],[3,2,1,-1],[1,1,-1,-2],[-1,-1,-2,-3]] -- 8
    print $ countNegatives [[3,2],[1,0]] -- 0

File: leetcode/1351. Count Negative Numbers in a Sorted Matrix/1351.js
// https://leetcode.com/problems/count-negative-numbers-in-a-sorted-matrix/
const _plus = (e, i) => e + i;
const countNegatives = (grid) => grid.map(e => e.filter(e => e < 0).length).reduce(_plus);

// Tests
countNegatives([[4,3,2,-1],[3,2,1,-1],[1,1,-1,-2],[-1,-1,-2,-3]]) // 18
countNegatives([[3,2],[1,0]]) // 0
